lib/upload.js

```diff
@@ -418,6 +419,9 @@ class Upload {
 
     let xhr = newRequest();
 
+    if (this.options.agent)
+      xhr.setAgent(this.options.agent);
```

Comment:
Please use curly braces to surround the if-statement's body.

Reply:
Also, please add a check that the `setAgent` method actually exists, which is not the case for browser environments. If it's not available, an error should be thrown.